### PR TITLE
Create SaaS App Template With Auth

### PR DESCRIPTION
**Pull request by julian-mcnichols**
* Wires up authentication, session, and telemetry in generic application project
* Alias relevant secret header
* Encapsulated pattern retained for project, no common project references utilized
* Include comments to highlight manual alterations to admin client
* Include comment to highlight developer task for their application secret prefix
* Clear meaningless links on BadgeMeUp SaaS page
* Forward reference to IAdminServiceClient within TenantService
* Leverage file scope namespaces in new app, retain code behind structure

### CODE REVIEW

File: src/Saas.Application/Saas.Application.Web/Models/ApplicationUser.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+namespace Saas.Application.Web.Models;
+
+public class ApplicationUser : ClaimsIdentity, IApplicationUser
+{
+    private static ClaimsIdentity Identity
+    {
+        get
+        {
+            return AppHttpContext.Current.User.Identity as ClaimsIdentity;
+        }
```
Comment (landonpierce) [inline src/Saas.Application/Saas.Application.Web/Models/ApplicationUser.cs:14]: Fix warnings

File: src/Saas.Application/Saas.Application.Web/Utilities/JsonSessionPersistenceProvider.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        public void Persist(string key, object value)
+        {
+            Session.SetString(key, JsonConvert.SerializeObject(value));
+        }
+
+        public T Retrieve<T>(string key)
+        {
+            var value = Session.GetString(key);
+
+            return value == null ? default(T) : JsonConvert.DeserializeObject<T>(value);
```
Comment (landonpierce) [inline src/Saas.Application/Saas.Application.Web/Utilities/JsonSessionPersistenceProvider.cs:25]: Fix warning

File: src/Saas.Application/Saas.Application.Web/Views/Shared/_LoginPartial.cshtml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿@using Microsoft.Identity.Web
+@if (User.Identity.IsAuthenticated)
```
Comment (landonpierce) [inline src/Saas.Application/Saas.Application.Web/Views/Shared/_LoginPartial.cshtml:2]: Fix warning

Comment (julian-mcnichols) [reply src/Saas.Application/Saas.Application.Web/Views/Shared/_LoginPartial.cshtml:2]: Per discussion, these are symptoms of a wider refactor that will affect the Signup app as well and will be handled in their own PR.

File: src/Saas.Application/Saas.Application.Web/Program.cs [hunk 4]
```diff
@@ -19,31 +72,40 @@
...
 // Add services to the container.
 var app = builder.Build();
 
 // Configure the HTTP request pipeline.
-if (!app.Environment.IsDevelopment())
+if (app.Environment.IsDevelopment())
 {
-    app.UseExceptionHandler("/Error");
+    app.UseDeveloperExceptionPage();
+    IdentityModelEventSource.ShowPII = true;
```
[the remark this answers is not in the document]
Comment (julian-mcnichols) [reply src/Saas.Application/Saas.Application.Web/Program.cs]: Updated PR

**Review by landonpierce — APPROVED** [review]

**Pull request merged**